fearless/Common/View/TableViewCell/SwipableTableViewCell.swift

```diff
@@ -0,0 +1,474 @@
+import Foundation
+import UIKit
+import SwiftUI
+
+protocol SwipeButtonProtocol: UIButton {
+    var type: SwipeButtonType { get }
+}
+
+enum SwipeButtonType: Int, CaseIterable {
+    case send
+    case receive
+    case teleport
+    case hide
+}
+
+protocol SwipeTableCellDelegate: AnyObject {
+    func swipeCellDidTap(on actionType: SwipeButtonType, with indexPath: IndexPath?)
+}
+
+// swiftlint:disable type_body_length file_length
+class SwipableTableViewCell: UITableViewCell {
+    // MARK: - Public properties
+
+    weak var delegate: SwipeTableCellDelegate?
+
+    /// The content view of a SwipableTableViewCell object is the default superview for content that the cell displays.
+    /// If you want to customize cells by simply adding additional views, you should add them to the content view so
+    /// they position appropriately as the cell transitions in to and out of editing mode.
+    let cloudView = UIView()
+    /// Backgrount view for action swipe menu from left side
+    var leftMenuBackgroundView = TriangularedBlurView()
+    /// Backgrount view for action swipe menu from right side
+    var rightMenuBackgroundView = TriangularedBlurView()
+
+    var cloudViewEdgeInsets: UIEdgeInsets {
+        UIEdgeInsets(
+            top: UIConstants.defaultOffset,
+            left: UIConstants.bigOffset,
+            bottom: 0,
+            right: UIConstants.bigOffset
+        )
+    }
+
+    var actionButtonWidht: CGFloat {
+        UIConstants.swipeTableActionButtonWidth
+    }
+
+    var openMenuTrigger: CGFloat {
+        actionButtonWidht * 0.7
+    }
+
+    var rightMenuButtons: [SwipeButtonProtocol] = []
+
+    var leftMenuButtons: [SwipeButtonProtocol] = []
+
+    /// Time for closing swipe action menu
+    var animationDuration: TimeInterval {
```

Comment:
you can name this variable "closeSwipeAnimationDuration" and use it without description comment. Don't thank me. I'm just doing my job.